### DISCUSSION TITLE
Some tabs not shown on old version of Android

### DISCUSSION THREAD

**Discussion by tantalate** [FossifyOrg/General-Discussion#569] [Q&A]
I have installed Fossify Player on an android watch, which runs android 8.1, but some of the tabs are missing (including Folders tab which is I really need). 

The same apk file running on new version of android have that tabs.

Is this because of the limitation of SDK version or permission settings?

**Comment by Aga-C** [comment]
When this feature was added, it was originally made only for Android 10+.

@naveensingh Do you know why it has been limited this way?

**Reply by naveensingh** [reply]
The code comment says the following: 

> // show Folders tab only on Android Q+, BUCKET_DISPLAY_NAME hasn't been available before that

The current implementation is simple and just groups the tracks by this bucket display name (i.e. folder name) and displays them in the folders tab. I suspect it was disabled on older versions to keep things 'simple'.

**Reply by naveensingh** [reply]
Folders tab can be enabled on older versions, we'll have to derive the folder name using the file path.
